gateway/ln-gateway/src/lib.rs

```diff
@@ -1313,6 +1313,27 @@ impl Gateway {
         Ok(channels)
     }
 
+    /// Withdraws funds from the gateway's lightning node on-chain wallet.
+    pub async fn handle_withdraw_onchain_msg(
+        &self,
+        WithdrawOnchainPayload {
+            address,
+            amount: amount_sats,
+            fee_rate_sats_per_vbyte,
+        }: WithdrawOnchainPayload,
+    ) -> Result<Txid> {
+        let context = self.get_lightning_context().await?;
+        let response = context
+            .lnrpc
+            .withdraw_onchain(
+                address.assume_checked(),
```

Comment:
Can we check the address here instead of using `assume_checked`? Not sure how cumbersome that will be

Reply:
Mind if I get around to this later? It does appear to be cumbersome. This work is being tracked through #5203 which I'm working on in the background.

Reply:
Sure